### PR TITLE
Fixing module numbering; handling pedestal runs

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

**Comment by shahor02** [comment]
@peressounko I get an exception in digitization with this patch:
````
o2-sim -n 5 -m CPV PHS -g pythia8hi --configKeyValues "Diamond.width[2]=6.; SimCutParams.maxRTracking=550; SimCutParams.maxAbsZTracking=300"
o2-sim-digitizer-workflow
...
[16992:PHOSDigitizer]: [11:43:11][ERROR] Exception caught: array::at: __n (which is 18446744073709550776) >= _Nm (which is 14337) 
[16991:CPVDigitizer]: [11:43:11][INFO] Digitization took 0.09s
[16992:PHOSDigitizer]: [11:43:11][STATE] Exiting FairMQ state machine
[16992:PHOSDigitizer]: [11:43:11][ERROR] Unhandled o2::framework::runtime_error reached the top of main, device shutting down. Recompile with DPL_ENABLE_BACKTRACE=1 to get more information.
````

File: Detectors/PHOS/workflow/src/ClusterizerSpec.cxx [hunk 1]
```diff
@@ -53,36 +52,17 @@ void ClusterizerSpec::run(framework::ProcessingContext& ctx)
...
-    printf("-------------Cells------------\n");
-    for (auto cTR : cellsTR) {
-      printf("new Event \n");
-      int firstCellInEvent = cTR.getFirstEntry();
-      int lastCellInEvent = firstCellInEvent + cTR.getNumberOfObjects();
-      for (int i = firstCellInEvent; i < lastCellInEvent; i++) {
-        Cell c = cells[i];
-        printf("cell[%d]: absId=%d, amp=%f, Time=%e \n", i, c.getAbsId(), c.getEnergy(), c.getTime());
-      }
+    std::unique_ptr<const o2::dataformats::MCTruthContainer<o2::phos::MCLabel>> truthcont;
```
Comment (peressounko) [inline Detectors/PHOS/workflow/src/ClusterizerSpec.cxx:59]: Dear @shahor02, do we copy list of primaries here or just move pointer? Is it correct way to access primaries?

**Comment by peressounko** [comment]
Dear @shahor02, thank you for spotting. I had to adjust some more files, sorry.
     Best regards, Dmitri

File: Detectors/PHOS/workflow/src/ClusterizerSpec.cxx (hunk 1 above)
Comment (shahor02) [reply Detectors/PHOS/workflow/src/ClusterizerSpec.cxx:59]: Hi @peressounko Yes, this is correct, accessing MCTruthContainer from the input will create internally a copy, you just access it via ``unique_ptr``. There is also a copy-less method, which requires different way of sending the message:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Steer/DigitizerWorkflow/src/ITSMFTDigitizerSpec.cxx#L180-L184

writing:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Detectors/ITSMFT/common/workflow/src/DigitWriterSpec.cxx#L85-L87

and reading:
https://github.com/AliceO2Group/AliceO2/blob/86f40deaceafe12d7638abd78679c19213457a25/Detectors/ITSMFT/ITS/workflow/src/ClustererSpec.cxx#L83-L88

Eventually, we should switch to these methods in all workflows.

**Review by shahor02 — CHANGES_REQUESTED** [review]
Hi @peressounko 
thanks, now the test passes, but the CI complains on narrowing conversions.

File: Detectors/PHOS/workflow/src/ClusterizerSpec.cxx (hunk 1 above)
Comment (peressounko) [reply Detectors/PHOS/workflow/src/ClusterizerSpec.cxx:59]: Thanks, I will try in next commit.

**Pull request merged**